spring-cloud-dataflow-admin-starter/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamDefinitionController.java

```diff
@@ -0,0 +1,251 @@
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PagedResourcesAssembler;
+import org.springframework.hateoas.ExposesResourceFor;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.hateoas.mvc.ResourceAssemblerSupport;
+import org.springframework.http.HttpStatus;
+import org.springframework.util.Assert;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Controller for operations on {@link StreamDefinition}. This
+ * includes CRUD and optional deployment operations.
+ *
+ * @author Eric Bottard
+ * @author Mark Fisher
+ * @author Patrick Peralta
+ * @author Ilayaperumal Gopinathan
+ */
+@RestController
+@RequestMapping("/streams/definitions")
+@ExposesResourceFor(StreamDefinitionResource.class)
+public class StreamDefinitionController {
+
+	private static final Logger logger = LoggerFactory.getLogger(StreamDefinitionController.class);
+
+	/**
+	 * The repository this controller will use for stream CRUD operations.
+	 */
+	private final StreamDefinitionRepository repository;
+
+	/**
+	 * The deployer this controller will use to deploy stream modules.
+	 */
+	private final ModuleDeployer deployer;
+
+	/**
+	 * Assembler for {@link StreamDefinitionResource} objects.
+	 */
+	private final Assembler streamAssembler = new Assembler();
+
+	private final StreamDeploymentController deploymentController;
+
+	/**
+	 * Create a {@code StreamController} that delegates
+	 * <ul>
+	 *     <li>CRUD operations to the provided {@link StreamDefinitionRepository}</li>
+	 *     <li>module coordinate retrieval to the provided {@link ArtifactRegistry}</li>
+	 *     <li>deployment operations to the provided {@link ModuleDeployer}</li>
+	 * </ul>
+	 *
+	 * @param repository  the repository this controller will use for stream CRUD operations
+	 * @param deploymentController    collabor
+	 * @param deployer    the deployer this controller will use to deploy stream modules
+	 */
+	@Autowired
+	public StreamDefinitionController(StreamDefinitionRepository repository, StreamDeploymentController deploymentController,
+	                                  @Qualifier("processModuleDeployer") ModuleDeployer deployer) {
+		Assert.notNull(repository, "repository must not be null");
+		Assert.notNull(deploymentController, "deploymentController must not be null");
+		Assert.notNull(deployer, "deployer must not be null");
+		this.deploymentController = deploymentController;
+		this.repository = repository;
+		this.deployer = deployer;
+	}
+
+	/**
+	 * Return a page-able list of {@link StreamDefinitionResource} defined streams.
```

Comment:
Return a page-able list of {@link StreamDefinitionResource} defined **in** streams.